docs/Releases.md

```diff
@@ -1,3 +1,74 @@
+- `publish-npm.js`: This is used by CircleCI, and is generally triggered by a new git tag. This script takes care of the actual `npm publish`, along with creating and publishing pre-build artifacts (for both iOS and Android). 
+  - For nightlies and dry-runs, it will call `set-rn-version` to bump versions in the repo.
+  - For releases (pre-release and stable), it is expected that CircleCI already ran `prepare-package-for-release.js` in an earlier job to bump versions.
+
+## How the React Native macOS Publish pipeline works
+
+### 0.68 and lower
+
+Our publish pipeline was mostly separate from React Native Core. At this point in time, we only re-used `set-rn-version.js`, with heavy modifications to:
+1. Add extra arguments to do the following:
+    - `rnmpublish` : git commit and tag the version bump
+    - `nightly` : Create a nightly build to be published off our main branch
+    - `autogenerate-version-number` : Autogenerate the next version number. Unlike React Native, we publish a new patch release on every commit via automation
+    - `skip-update-ruby` : This used to cause publish failures, so we added an arg to skip it. 
+2. Not destructively delete `private` / `workspace` keys from the package.json file (we had separate steps to delete and restore those keys in our pipeline)
+3. Make it more similar to `bump-oss-version.js` (the intention was to make it the one script to call that is more CI friendly)
+4. .. but also skip some of those modifications with the extra `rnmpublish` flag because we do `git tag` and `git push` separately
+
+The Publish flow does the following:
+
+1. Set tags for NPM based on the branch
+2. Conditional based on branch:
+    - If we're on the *main* branch
+      - Call `set-rn-version` with the extra nightly / rnm-publish args
+    - If we're on a *stable* branch
+      - Call our own script `bumpFileVersions` to auto-bump versions in files, calling `set-rn-version`
+4. Remove `workspace` / `private` keys from package.json manually 
+5. Publish to NPM
+6. Restore `workspace` / `private` keys from package.json manually 
+7. `gitTagRelease.js` to push the tags and new version bump back to git.
+
+### 0.71 and higher
+
+An attempt was made to simplify the steps above and re-use more of the scripts that React Native Core uses. Namely:
+- Use more of the RN scripts to handle things like git tagging, git pushing, and NPM publishing. The intention is to leverage new features that have been added to those scripts, like the ability to build nightlies and dry runs, along with increased safety via checks on the version number. 
+- Don't bother with manually removing and restoring workspace config. We don't need the `private` field set anyway since we don't have beachball auto-publishing or anything. 
+- Extract all the steps to a template `apple-job-publish` with a parameter to switch between nightlies, dry runs, and releases. This was we can now add a new "NPM Publish Dry Run" step to our PR checks.
```

Comment:
```suggestion
- Extract all the steps to a template `apple-job-publish` with a parameter to switch between nightlies, dry runs, and releases. This was done so that we can now add a new "NPM Publish Dry Run" step to our PR checks.
```